### ISSUE TITLE
Replace (r) does not work with accented characters on OSX

### ISSUE THREAD

**Issue by Valve** [atom/vim-mode#791]
In OSX you can type accented characters (éåñ) etc directly from keyboard. 
To type é, you first press alt+e, then e. Replacing the character with (r) should allow to type in the accented version of the character, but it doesn't.

Example:

I want to type México, so I type Mexico and then want to replace e with é. So I place the cursor above e and press (r) and then alt+e,e. 

Expected result: México
Actual result: M´xico.

**Comment by jacekkopecky** [comment]
Oh, this one is of interest to me - I'll try to get to it when we're done merging the current back log.

**Comment by bronson** [comment]
Wow, on my OSX it's even worse.  I `R` above an e, hit `alt-e` and `e` and crash!  Of course, everything works in insert mode.

**Atom Version**: 1.0.3
**System**: Mac OS X 10.10.4
**Thrown From**: Atom Core


### Stack Trace

Uncaught TypeError: Cannot read property 'split' of null

```
At /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:756

TypeError: Cannot read property 'split' of null
  at Selection.module.exports.Selection.insertText (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/selection.js:451:28)
  at /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/text-editor.js:736:29
  at /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/text-editor.js:778:29
  at TextBuffer.module.exports.TextBuffer.transact (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:751:18)
  at TextEditor.module.exports.TextEditor.transact (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1151:26)
  at /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/text-editor.js:772:24
  at TextEditor.module.exports.TextEditor.mergeSelections (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/text-editor.js:2017:18)
  at TextEditor.module.exports.TextEditor.mergeIntersectingSelections (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1987:35)
  at TextEditor.module.exports.TextEditor.mutateSelectedText (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/text-editor.js:770:19)
  at TextEditor.module.exports.TextEditor.insertText (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/text-editor.js:733:19)
  at TextEditor.object.(anonymous function) [as insertText] (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/underscore-plus/lib/underscore-plus.js:77:27)
  at HTMLDivElement.<anonymous> (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:363:24)

```

### Commands

```
     -9:54.7.0 editor:consolidate-selections (atom-text-editor.editor.mini.is-focused)
     -9:54.7.0 core:cancel (atom-text-editor.editor.mini.is-focused)
     -9:54.7.0 blur (atom-text-editor.editor.mini.is-focused)
     -9:54.5.0 vim-mode:reset-normal-mode (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
  6x -9:51.9.0 blur (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
  3x -0:08.0 vim-mode:move-left (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
     -0:06.0 vim-mode:replace (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
  3x -0:06.0 blur (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "metrics",
      "exception-reporting",
      "wrap-guide",
      "welcome",
      "background-tips",
      "linter-rubocop",
      "package-cop",
      "linter-haml",
      "smart-tab-name",
      "linter-eslint",
      "linter-jscs",
      "linter-jshint",
      "vim-mode"
    ],
    "themes": [
      "one-dark-ui",
      "oceanic-reef-syntax"
    ],
    "destroyEmptyPanes": false
  },
  "editor": {
    "invisibles": {},
    "autoIndentOnPaste": false
  }
}
```

### Installed Packages

```coffee
# User
atom-ternjs, v0.7.1
color-picker, v2.0.11
cursor-history, v0.4.15
file-icons, v1.6.2
git-log, v0.4.1
highlight-selected, v0.10.1
language-haml, v0.21.0
linter, v1.2.4
linter-coffeelint, v1.0.1
linter-csslint, v0.0.14
linter-erb, v0.1.1
linter-ruby, v0.1.6
oceanic-reef-syntax, v0.6.0
project-find-navigation, v0.0.9
react, v0.12.6
vim-mode-next, v0.57.0
vim-mode-visual-block, v0.2.13
vim-surround, v0.8.0
white-cursor, v0.6.0

# Dev
No dev packages
```

**Comment by bronson** [comment]
Note that this affects `r` too.  Might be a separate issue.  How would the replace operator know that it needs to wait for two characters to arrive?

**Comment by bronson** [comment]
Strange, after that crash, Atom seems to be mostly behaving.

Initial text: Mexico
Actual result: Méico
Expected result: México

So the only issue I'm seeing now is that replace mode replaced two characters with one.

@Valve did you type the final `e`?  I see your result after hitting `alt-e` and before hitting `e`.

**Comment by jacekkopecky** [comment]
It's an issue for the replace mode as well as for the `r` single replace
command.

**Comment by Valve** [comment]
@bronson , yes, I did, alt+e, then e.

**Comment by jacekkopecky** [comment]
@50Wliu this one no longer needs reproduction, it's been reproduced. :smiley:

**Comment by jacekkopecky** [comment]
This is an issue in Atom: atom/atom#5414 (affects `t`, `f`, `r`) and atom/atom#8486 (affects replace mode).

**Comment by jacekkopecky** [comment]
the replace mode problem will be (mostly?) solved by atom/atom#8526 when they :ship: the next version of Atom

the `r` issue also affects `f` and `m` and so on, users of the hidden variant of `VimNormalModeInputElement`, and is addressed in #846

**Comment by Valve** [comment]
@jacekkopecky thanks!
